### PR TITLE
Adding mutex locks in cache to prevent race conditions 

### PR DESCRIPTION
**Pull request by lucamrgs**

### CODE REVIEW

**Comment by lucamrgs** [comment]
@MaartendeKruijf how do we test this?

**Comment by lucamrgs** [comment]
@MaartendeKruijf is this enough or should we do some more in-depth testing and such?

**Comment by MaartendeKruijf** [comment]
@lucamrgs let met check

File: internal/reporter/downstream_reporter/cache/cache.go [hunk 1]
```diff
@@ -85,7 +68,43 @@ func (cacheReporter *Cache) addExecution(newExecutionEntry cache_report.Executio
 	return nil
 }
 
+func (cacheReporter *Cache) GetExecutions() ([]cache_report.ExecutionEntry, error) {
+	cacheReporter.mutex.Lock()
```
Comment (MaartendeKruijf) [inline internal/reporter/downstream_reporter/cache/cache.go:72]: I would gaurd the most critical sections in getExecutions and addExecutions. Also then just use a go routine to let the decomposer not wait on any mutex.

**Review by MaartendeKruijf — CHANGES_REQUESTED** [review]

**Comment by lucamrgs** [comment]
@MaartendeKruijf both requested changes are implemented

**Review by MaartendeKruijf — APPROVED** [review]

**Pull request merged**